internal/envir/util.go

```diff
@@ -43,6 +39,28 @@ func IsCI() bool {
 	return ci && err == nil
 }
 
+func ShellStart() time.Time {
+	return ParseShellStart(os.Getenv(DevboxShellStartTime))
+}
+
+func FormatShellStart(t time.Time) string {
+	if t.IsZero() {
+		return ""
+	}
+	return strconv.FormatInt(t.Unix(), 10)
+}
+
+func ParseShellStart(s string) time.Time {
+	if s == "" {
+		return time.Time{}
+	}
+	unix, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return time.Time{}
+	}
+	return time.Unix(unix, 0)
+}
```

Comment:
I'd prefer keeping this logic out of the `envir` package to keep it streamlined. 

I'd suggest it is better encapsulated in the telemetry package, or even a more focussed `shtelemetry` package